src/Http/Faking/Fixture.php

```diff
@@ -46,7 +63,33 @@ public function getMockResponse(): ?MockResponse
         $fixturePath = $this->getFixturePath();
 
         if ($storage->exists($fixturePath)) {
-            return RecordedResponse::fromFile($storage->get($fixturePath))->toMockResponse();
+            $response = RecordedResponse::fromFile($storage->get($fixturePath))->toMockResponse();
+
+            if (! is_array($this->merge)) {
+                return $response;
+            }
+
+            // First, we get the body as an array. If we're dealing with
+            // a `StringBodyRepository`, we have to encode it first.
+            if (! is_array($body = $response->body()->all())) {
+                $body = json_decode($body ?: '[]', associative: true, flags: \JSON_THROW_ON_ERROR);
+            }
```

Comment:
I think this might be a breaking change for fixtures that are just a string, and not necesarily JSON. I have fixtures in the past which is just plain text. Please could you add a test for checking plain-text fixtures to ensure this isn't a breaking change?

Additionally, please could we allow so people can use recursive arrays **or** dot-notation e.g

```php
MockResponse::fixture(...)->merge([
      'person' => [
            'name' => 'Sam'
      ],
]);
```

Reply:
> @Sammyjo20
> I think this might be a breaking change for fixtures that are just a string, and not necesarily JSON.

Gotta admit I'm not sure how to have plain-text fixtures, they automatically save themselves as `.json` and `RecordedResponse#toFile` uses `json_encode`. Could you hint me where to look at to write this test?

That being said, there is a return statement above that condition that checks for the presence of `merge`, so there shouldn't be a breaking change.

&nbsp;


> @Sammyjo20
> Additionally, please could we allow so people can use recursive arrays **or** dot-notation

I've used Laravel's `Arr::set`, which doesn't support that. `data_set` doesn't support it either. We would have to use `array_merge_recursive` or rewrite the function ourselves.

But honestly, that would lead to some unexpected results, eg. appending data to arrays instead of replacing them. This is what happened to me when I first used `array_merge_recursive` when dogfeeding this feature.

A more flexible option would be to allow passing a callback:

```php
MockResponse::fixture(...)->merge(function (array $data) {
    // do whatever with data?
    return array_merge_recursive($data, [
        'person' => [
            'name' => 'Sam'
        ],
    ]);
});
```

Reply:
I just updated the implementation to add support for closures. If you don't want that, I can revert the commit, no problem 👍